### PR TITLE
m1 support for go

### PR DESCRIPTION
**Pull request by saolsen**

### CODE REVIEW

**Review by gj — CHANGES_REQUESTED** [review]
Can we now remove `languages/go/internal/ffi/macos/empty.go`?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -532,6 +534,32 @@ jobs:
...
+        uses: actions/download-artifact@v1
+        with:
+          name: go
+      # go 1.16 is the earliest version supported on m1
+      - name: "test"
+        run: |
+          ~/go/bin/go1.16 build
+          ~/go/bin/go1.16 test -v ./tests/ 
+          ~/go/bin/go1.17 build
+          ~/go/bin/go1.17 test -v ./tests/
```
[the remark this answers is not in the document]
Comment (gj) [reply .github/workflows/release.yml:556]: nit: Another nice thing about using `strategy.matrix.go-version` is it'd make it clear which test logs are from 1.16 and which are from 1.17. I looked at [the successful run](https://github.com/osohq/oso/runs/4194618804?check_suite_focus=true) to make sure the new code works and saw duplicate test logs, but it wasn't clear when the 1.16 tests end and when the 1.17 tests begin.

File: languages/go/Makefile [hunk 2]
```diff
@@ -12,7 +12,7 @@ ifeq ($(UNAME_S),Linux)
 	cp ../../target/debug/libpolar.a internal/ffi/native/linux/
 endif
 ifeq ($(UNAME_S),Darwin)
-	cp ../../target/debug/libpolar.a internal/ffi/native/macos/
+	cp ../../target/debug/libpolar.a internal/ffi/native/macos/amd64/
```
[the remark this answers is not in the document]
Comment (saolsen) [reply languages/go/Makefile:15]: yeah

**Comment by saolsen** [comment]
> Can we now remove `languages/go/internal/ffi/macos/empty.go`?

nope. need it apparently or go doesn't realize the subfolders are part of the module.

**Review by gj — APPROVED** [review]

**Pull request merged**